### ISSUE TITLE
regolith-control-center crash

### ISSUE THREAD

**Issue by ploum** [regolith-linux/regolith-control-center#2]
version 1.41.9-1regolith is crashing
stderr : 
[cc.txt](https://github.com/regolith-linux/regolith-control-center/files/8473561/cc.txt)

stdout :  

Bail out! ERROR:../shell/cc-shell-model.c:419:cc_shell_model_set_panel_visibility: assertion failed: (valid)

**Comment by kgilmer** [comment]
Possibly related (from slack):

```
Just tried regolith-control-center and get the following error (on Pop_OS! 22.04 Beta with X and Intel/Nvidia Hybrid mode):
➤ regolith-control-center
../src/intel/isl/isl.c:2212: FINISHME: ../src/intel/isl/isl.c:isl_surf_supports_ccs: CCS for 3D textures is disabled, but a workaround is available.
** (regolith-control-center:130801): WARNING **: 10:45:46.272: Ignoring broken panel diagnostics (missing desktop file)
**
ERROR:../shell/cc-shell-model.c:419:cc_shell_model_set_panel_visibility: assertion failed: (valid)
Bail out! ERROR:../shell/cc-shell-model.c:419:cc_shell_model_set_panel_visibility: assertion failed: (valid)
fish: Job 1, 'regolith-control-center' terminated by signal SIGABRT (Abort)
```

**Comment by karl-run** [comment]
Running on Pop!_OS 22.04, `regolith-control-center 41.4`, when running from the terminal:

```
➜  ~ regolith-control-center -v
** (regolith-control-center:36530): WARNING **: 21:22:41.302: Ignoring broken panel diagnostics (missing desktop file)
** (regolith-control-center:36530): ERROR **: 21:22:41.304: Error in cc_shell_model_set_panel_visibility, unknown panel id: diagnostics
[1]    36530 trace trap (core dumped)  regolith-control-center -v
```

**Comment by TS-CUBED** [comment]
This error seems to be Pop_OS! specific. Pop_OS! has made some changes to gnome-control-center and I assume Regolith does not address these differences when setting up regolith-control-center.

**Comment by ploum** [comment]
Error happens on a straight Ubuntu 22.04

**Comment by dhlbrg** [comment]
I got the same issue as karl-run on Pop!_ OS 22.04

**Comment by kgilmer** [comment]
I cannot reproduce this, any hints?

On Jammy, I've tried manually setting `last-panel` to `diagnostics` but it does not result in a crash on version `1.41.11` via `gsettings set org.gnome.ControlCenter last-panel "diagnostics"`.

It seems that there may be a codepath from dbus that's causing the crash but unsure of how to trigger it.  I need to repro on Ubuntu.

**Comment by ploum** [comment]
I’ve reproduced this bug with an empty $home, which means that the issue is not related to any user preference.

Is there a way to help with some meaningful trace? I could give some strace output if I knew which arguments were useful in this case.

**Comment by kgilmer** [comment]
Are you just running `regolith-control-center` and it directly crashes?  Any steps I can reproduce?  See the commit above for a possible fix, are you able to build and verify?  Should I build and send you a binary for testing?

**Comment by ploum** [comment]
yep, running directly and it crashes. You could either send me the instructions to build it or a binary, whatever is easier for you (no idea how to build this thing).  I’m just trying to not mess with the / system (so no make install) and building it in a local folder.

**Comment by kgilmer** [comment]
I was able to build in `jammy` with the following steps:
(unfortunately there are a lot of patches in the debian dir so it's not possible to simply build directly w/ meson, we have to use the deb tools so that patches are applied)

```
git clone git@github.com:regolith-linux/regolith-control-center.git
cd regolith-control-center/
git checkout invalid-panel-fix-jammy
sudo apt build-dep -y .  # This will install all build-scope package dependencies
debuild -i -us -uc -b  # debuild comes from the package devscripts
debian/regolith-control-center/usr/bin/regolith-control-center # this is the built test target
```

**Comment by revkarol** [comment]
I eventually got this building with jammy (on Pop OS 22.04) - a lot of complaints about positional arguments in `i18n.merge_file` from meson - I guess that's a version thing (I have meson 0.61.2).  Then after building I got the same error as @ploum (which I also get when running the standard apt installed version).  I hacked away the assert on line 427 of `shell/cc-shell-model.c` 
Then it ran ok, until I opened the mouse tab and it died again  with the error 
```Settings schema 'org.gnome.settings-daemon.peripherals.mouse' is not installed```.  I can't really spot many differences between this variant and the Pop OS version (apart from the crashing obviously).  So I can live without it myself (even if I don't get to see the nice Regolith logo in my about tab :disappointed: ).  I'm just hacking about without much expertise here.  Otherwise beta2 is working pretty well for me :smiley:  Hope this is helpful.

**Comment by kgilmer** [comment]
Thanks for spending the time to do the repro @revkarol , very helpful.  I've been able to install popos in a VM and see the problem so hopefully I can figure out a fix shortly.

**Comment by kgilmer** [comment]
Potential fix building in https://github.com/regolith-linux/voulage/actions/runs/2382065776

To test this fix you'll need to update your Regolith repo from `release` to `testing`.  To do this, update the s3 url in the apt sources file and replace `release` with `testing`.

```
$ cat /etc/apt/sources.list.d/regolith.list 
deb [arch=amd64] https://regolith-release-debian-bullseye-amd64.s3.amazonaws.com bullseye main
```

to

```
$ cat /etc/apt/sources.list.d/regolith.list 
deb [arch=amd64] https://regolith-testing-debian-bullseye-amd64.s3.amazonaws.com bullseye main
```

Keep in mind that the rest of the source is likely different than what I have here.  The only part that changes is `release` to `testing`.

**Comment by servzin** [comment]
The fix from the `testing` branch worked for me on Pop_OS! 22.04. Thank you @kgilmer for digging down into the issue, I highly appreciate it!

**Comment by t3st-account** [comment]
I also can reproduce this issue in Linux Mint 20.3 (based on focal)
the testing branch does not resolve the issue for me

**Comment by t3st-account** [comment]
as of regolith-control-center 3.36.5 the bug is fixed on Linux Mint 20.3
